src/p2p/net_peerlist.h

```diff
@@ -81,6 +81,8 @@ namespace nodetool
     bool set_peer_just_seen(peerid_type peer, const net_address& addr);
     bool set_peer_unreachable(const peerlist_entry& pr);
     bool is_ip_allowed(uint32_t ip);
+    bool get_gray_peer_random(peerlist_entry& pe);
+    bool remove_from_peer_gray(const peerlist_entry& pe);
```

Comment:
A bit bikeshedding, but I'd rather names that make sense, eg, remove_from_gray_peerlist (or remove_from_peerlist with the actual peerlist as arg), also get_random_gray_peer. It just makes it easier to follow I think.